### PR TITLE
DuckDB Implementation

### PR DESCRIPTION
**Pull request by yilinxia**
We aim to add duckdb into Logica as an alternative query engine
This PR records the implementation details

- Add DuckDB connect
- Translate 
   - argmin, argmax, argmink, argmaxk
   - list and array

### CODE REVIEW

File: colab_logica.py [hunk 1]
```diff
@@ -309,17 +317,21 @@ def Logica(line, cell, run_query):
                 color.Warn(predicate + '_sql'))
 
   with bar.output_to(logs_idx):
+    print(sql)
```
Comment (EvgSkv) [inline colab_logica.py:320]: Can you please remove the print? :-)

File: common/duckdb_logica.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+from decimal import Decimal
+import sqlite3
+
+if '.' not in __package__:
+  from type_inference.research import infer
+else:
+  from ..type_inference.research import infer
+
+
+def PostgresExecute(sql, connection):
```
Comment (EvgSkv) [inline common/duckdb_logica.py:30]: This looks like a copy of postgres file. Do we need it at all?

File: colab_logica.py (hunk 1 above)
Comment (yilinxia) [reply colab_logica.py:320]: That's for debugging purpose. Will remove it once we finish passing all tests

File: common/duckdb_logica.py (hunk 2 above)
Comment (yilinxia) [reply common/duckdb_logica.py:30]: I think we don't need that and I will remove them.

**Comment by EvgSkv** [comment]
Excellent! Thank you for your contribution!

**Pull request merged**